На проверку ДЗ0

red1r3ct opened the pull request:
Реализовано логирование сервера

Commit:
A lot of changes, libs, router, require thing

Добавлена реализация модульной структуры, бэкбоновский роутер для
менюшки, скачаны либы, начаты вьюхи. Код требует переработки

Commit:
Main View

Вьюга главной страницы

Commit:
Screen hide added

Добавлен вызов скрытия предыдущего экрана (.hide()) до показа нового
(.show()).

red1r3ct committed:
Merge pull request #1 from pewpew-team/menu-logic

Menu logic

Commit:
Commit

Commit:
Clickable

Добавлено прослушивание событий нажатия на кнопки меню

burlakilia commented on Gruntfile.js:
```diff
@@ -21,7 +21,7 @@ module.exports = function (grunt) {
                     template: function (data) {
                         return grunt.template.process(
-                            'var <%= name %>Tmpl = <%= contents %> ;',
+                            'define(function () { return <%= contents %> ; });',
```
зачем ?

red1r3ct replied:
Для того что бы фестовые шаблоны работали вместе в require.js, то есть были
модулями.

вторник, 16 февраля 2016 г. пользователь Burlak Ilia написал:

> In Gruntfile.js
> https://github.com/frontend-park-mail-ru/2016_1_PewPew/pull/2#discussion_r52973127
> :
> 
> > @@ -21,7 +21,7 @@ module.exports = function (grunt) {
> >                  options: {
> >                      template: function (data) {
> >                          return grunt.template.process(
> > -                            'var <%= name %>Tmpl = <%= contents %> ;',
> > -                            'define(function () { return <%= contents %> ; });',
> 
> зачем ?
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/frontend-park-mail-ru/2016_1_PewPew/pull/2/files#r52973127
> .

burlakilia replied:
тогда норм

burlakilia commented on public_html/js/router.js:
```diff
@@ -0,0 +1,38 @@
+            initialize: function () {
+                this.currentView = mainView;
+            },
+            scoreboardAction: function () {
```
Мне одному кажется, что здесь нарушен принцип DRY ?
Как это сделать без копипаста ?

red1r3ct replied:
Могу предложить такой вариант https://github.com/pewpew-team/2016_1_PewPew/blob/master/public_html/js/router.js

Commit:
Merge branch 'master' into menu-logic

Commit:
Router changed according to DRY

red1r3ct committed:
Merge pull request #3 from pewpew-team/menu-logic

Menu logic

Commit:
Advanced server logs

Commit:
\login POST handling added

Сервер отвечает на POST запрос по URL /login и переданными username и
password токеном по которому будет производится все дальнейшее
взаимодействие

red1r3ct committed:
Merge pull request #4 from pewpew-team/node-server

Node server

Commit:
A lot of changes

Навигация осуществляется с помощью Backbone.Events. Добавлена базовая
вьюха, остальные ее расширяют методом extend(). Всем кнопкам при показе
добавляются ивенты в соответствии с их id.

Commit:
Изменения в верстке

Добавлен экран регистрации, другие экраны переработаны

Commit:
Scoreboard screen basic

Commit:
Body-parser added

red1r3ct committed:
Merge pull request #5 from pewpew-team/auth

Auth

Commit:
jquery request to server

Добавлен метод отправки сообщения на сервер для входа.

itwillwork committed:
доделано меню

Commit:
Request to /login

Добавлен метод модели юзера запрашивающий токен по логину и паролю

itwillwork committed:
test

Commit:
Game merged with menu

Commit:
Start refactoring

Commit:
Router refactored

Подумать как убрать повторяющийся фрагмент в объявление роутов

Commit:
Some changes

Commit:
Merge branch 'refactoring' into basic-design

Conflicts:
	public_html/js/router.js

Commit:
Merge branch 'basic-design' into refactoring

Commit:
listen to start game

red1r3ct committed:
Merge pull request #7 from pewpew-team/refactoring

Refactoring

Commit:
Rename

red1r3ct committed:
Merge pull request #8 from pewpew-team/auth

Auth

Commit:
Game route added
